core-bundle/tests/Contao/PageModelTest.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Contao;
+
+use Contao\CoreBundle\Security\Authentication\Token\TokenChecker;
+use Contao\Database;
+use Contao\Database\Result;
+use Contao\Database\Statement;
+use Contao\Model\Collection;
+use Contao\Model\Registry;
+use Contao\PageModel;
+use Contao\System;
+use Contao\TestCase\ContaoTestCase;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Symfony\Component\Filesystem\Filesystem;
+
+class PageModelTest extends ContaoTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $platform = $this->createMock(AbstractPlatform::class);
+        $platform
+            ->method('getIdentifierQuoteCharacter')
+            ->willReturn('\'')
+        ;
+
+        $connection = $this->createMock(Connection::class);
+        $connection
+            ->method('getDatabasePlatform')
+            ->willReturn($platform)
+        ;
+
+        $connection
+            ->method('quoteIdentifier')
+            ->willReturnArgument(0)
+        ;
+
+        $container = $this->getContainerWithContaoConfiguration();
+        $container->set('database_connection', $connection);
+        $container->set('contao.security.token_checker', $this->createMock(TokenChecker::class));
+        $container->setParameter('contao.resources_paths', $this->getTempDir());
+
+        (new Filesystem())->mkdir($this->getTempDir().'/languages/en');
+
+        System::setContainer($container);
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+
+        Registry::getInstance()->reset();
+
+        // Reset database instance
+        $property = (new \ReflectionClass(Database::class))->getProperty('arrInstances');
+        $property->setAccessible(true);
+        $property->setValue([]);
```

Comment:
We should probably add a `reset` method to the `Database` class as well. Wdyt?
(Or I'll do it in #2518 - that one's still in progress because things turned out way more complicated than I thought)

Reply:
Yes, adding it to #2518 would make sense I think.